azure-core/azure/core/pipeline/policies/retry_async.py

```diff
@@ -85,12 +85,14 @@ class AsyncRetryPolicy(RetryPolicy, AsyncHTTPPolicy):
                     if retries_remaining:
                         await self.sleep(retry_settings, request.context.transport, response=response)
                         continue
-                return response
+                break
             except AzureError as err:
                 if self._is_method_retryable(retry_settings, request.http_request):
                     retries_remaining = self.increment(retry_settings, response=request, error=err)
                     if retries_remaining:
                         await self.sleep(retry_settings, request.context.transport)
                         continue
                 raise err
+        if retry_settings['history']:
+            response.context['history'] = retry_settings['history']
```

Comment:
I like to encapsulate dict manipulations with raw string keys behind functions when I can. Since this is duplicated in `RetryPolicy` (and also since this inherits from `RetryPolicy`), might consider putting these two lines in a method `self.update_context(context, retry_settings)`

Reply:
Will add issue to track this.